aws/resource_aws_s3_object_copy.go

```diff
@@ -0,0 +1,639 @@
+package aws
+
+import (
+	"fmt"
+	"net/url"
+	"strings"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/s3"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+	"github.com/terraform-providers/terraform-provider-aws/aws/internal/keyvaluetags"
+)
+
+func resourceAwsS3ObjectCopy() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceAwsS3ObjectCopyCreate,
+		Read:   resourceAwsS3BucketObjectRead,
+		Update: resourceAwsS3ObjectCopyUpdate,
+		Delete: resourceAwsS3BucketObjectDelete,
```

Comment:
Typically, we avoid reusing implementations across resource types. `resourceAwsS3BucketObjectRead` and `resourceAwsS3BucketObjectDelete` should have local implementations

Reply:
Added